src/Bicep.LangServer/Completions/BicepCompletionContext.cs

```diff
@@ -153,70 +153,54 @@ private static List<SyntaxBase> FindNodesMatchingOffset(ProgramSyntax syntax, in
 
         private static BicepCompletionContextKind GetDeclarationTypeFlags(IList<SyntaxBase> matchingNodes, int offset)
         {
-            if (matchingNodes.Count < 2)
+            // local function
+            bool CheckTypeIsExpected(SyntaxBase name, SyntaxBase type) => name.Span.Length > 0 && offset > name.Span.Position + name.Span.Length && offset <= type.Span.Position;
```

Comment:
The fix was to adjust the condition in this function (the commit called "fix" covers it... mostly). All other changes in this file are just a refactor to change this logic to use the tail matcher.